src/terms/local.jl

```diff
@@ -82,78 +82,133 @@ end
     end
 
-    enforce_real!(basis, pot_fourier)  # Symmetrize Fourier coeffs to have real iFFT
-    pot_real = irfft(basis, to_device(basis.architecture, pot_fourier))
-
-    TermAtomicLocal(pot_real)
+    iszero(q) && enforce_real!(basis, pot_fourier)  # Symmetrize coeffs to have real iFFT
+    ifft(S, basis, to_device(basis.architecture, pot_fourier))
 end
+(::AtomicLocal)(basis::PlaneWaveBasis{T}) where {T} =
+    TermAtomicLocal(compute_local_potential(T, basis))
 
-@timing "forces: local" function compute_forces(::TermAtomicLocal, basis::PlaneWaveBasis{TT},
-                                                ψ, occupation; ρ, kwargs...) where {TT}
-    T = promote_type(TT, real(eltype(ψ[1])))
+function compute_forces(term::TermAtomicLocal, basis::PlaneWaveBasis{T}, ψ, occupation;
+                        ρ, kwargs...) where {T}
+    S = promote_type(T, real(eltype(ψ[1])))
+    forces_local(S, basis, ρ, zero(Vec3{T}))
+end
+@timing "forces: local" function forces_local(S, basis::PlaneWaveBasis{T}, ρ, q) where {T}
     model = basis.model
+    recip_lattice = model.recip_lattice
     ρ_fourier = fft(basis, total_density(ρ))
 
     # energy = sum of form_factor(G) * struct_factor(G) * rho(G)
     # where struct_factor(G) = e^{-i G·r}
-    forces = [zero(Vec3{T}) for _ = 1:length(model.positions)]
+    forces = [zero(Vec3{S}) for _ = 1:length(model.positions)]
     for group in model.atom_groups
         element = model.atoms[first(group)]
-        form_factors = [Complex{T}(local_potential_fourier(element, norm(G)))
-                        for G in G_vectors_cart(basis)]
+        form_factors = [complex(S)(local_potential_fourier(element, norm(recip_lattice * (G + q))))
+                        for G in G_vectors(basis)]
         for idx in group
             r = model.positions[idx]
-            forces[idx] = _force_local_internal(basis, ρ_fourier, form_factors, r)
+            forces[idx] = -force_type(S, sum(conj(ρ_fourier[iG])
+                                                  * form_factors[iG]
+                                                  * cis2pi(-dot(G + q, r))
+                                                  * (-2T(π)) * (G + q) * im
+                                                  / sqrt(model.unit_cell_volume)
+                                             for (iG, G) in enumerate(G_vectors(basis))))
         end
     end
     forces
 end
 
-# function barrier to work around various type instabilities
-function _force_local_internal(basis, ρ_fourier, form_factors, r)
-    T = real(eltype(ρ_fourier))
-    f = zero(Vec3{T})
-    for (iG, G) in enumerate(G_vectors(basis))
-        f -= real(conj(ρ_fourier[iG])
-                  .* form_factors[iG]
-                  .* cis2pi(-dot(G, r))
-                  .* (-2T(π)) .* G .* im
-                  ./ sqrt(basis.model.unit_cell_volume))
+# Phonon: Perturbation of the local potential with respect to a displacement on the
+# direction α of the atom s.
```

Comment:
That's a helpful comment for example (along the lines of commenting `compute_δH\psi_sa`)